Disable torch.complex32 dtype

anjali411 opened the pull request:
Related https://github.com/pytorch/pytorch/issues/69209#event-5722454832

Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #70606



cc @ezyang @gchanan @anjali411 @dylanbespalko @mruberry @Lezcano @nikitaved

Differential Revision: [D33954861](https://our.internmc.facebook.com/intern/diff/D33954861)

anjali411 committed:
Disable torch.complex32 dtype

[ghstack-poisoned]

anjali411 commented:
@ngimel should we just cherry-pick this for the next release?

anjali411 committed:
Update on "Disable torch.complex32 dtype"




[ghstack-poisoned]

mruberry commented on torch/testing/_comparison.py:
```diff
@@ -1106,8 +1105,6 @@ def assert_close(
     +---------------------------+------------+----------+
```
cc @pmeier

mruberry reviewed (approved):
Cool! (So long as the tests pass.)

Note from offline meeting: @ngimel plans to file a follow-up issue for proper ComplexHalf support

anjali411 committed:
Update on "Disable torch.complex32 dtype"

Related https://github.com/pytorch/pytorch/issues/69209#event-5722454832




cc ezyang gchanan anjali411 dylanbespalko mruberry @Lezcano nikitaved

[ghstack-poisoned]

anjali411 committed:
Update on "Disable torch.complex32 dtype"

Related https://github.com/pytorch/pytorch/issues/69209#event-5722454832




cc ezyang gchanan anjali411 dylanbespalko mruberry @Lezcano nikitaved

[ghstack-poisoned]

anjali411 commented:
@pytorchbot ciflow rerun -l ciflow/all

anjali411 committed:
Update on "Disable torch.complex32 dtype"

Related https://github.com/pytorch/pytorch/issues/69209#event-5722454832




cc ezyang gchanan anjali411 dylanbespalko mruberry @Lezcano nikitaved

[ghstack-poisoned]

anjali411 committed:
Update on "Disable torch.complex32 dtype"

Related https://github.com/pytorch/pytorch/issues/69209#event-5722454832




cc ezyang gchanan anjali411 dylanbespalko mruberry @Lezcano nikitaved

[ghstack-poisoned]

anjali411 commented:
@anjali411 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33954861).

anjali411 committed:
Update on "Disable torch.complex32 dtype"

Related https://github.com/pytorch/pytorch/issues/69209#event-5722454832




cc ezyang gchanan anjali411 dylanbespalko mruberry @Lezcano nikitaved

Differential Revision: [D33954861](https://our.internmc.facebook.com/intern/diff/D33954861)

[ghstack-poisoned]

anjali411 commented:
@anjali411 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33954861).

anjali411 committed:
Update on "Disable torch.complex32 dtype"

Related https://github.com/pytorch/pytorch/issues/69209#event-5722454832




cc ezyang gchanan anjali411 dylanbespalko mruberry @Lezcano nikitaved

Differential Revision: [D33954861](https://our.internmc.facebook.com/intern/diff/D33954861)

[ghstack-poisoned]

anjali411 committed:
Update on "Disable torch.complex32 dtype"

Related https://github.com/pytorch/pytorch/issues/69209#event-5722454832




cc ezyang gchanan anjali411 dylanbespalko mruberry @Lezcano nikitaved

Differential Revision: [D33954861](https://our.internmc.facebook.com/intern/diff/D33954861)

[ghstack-poisoned]

anjali411 commented:
@JackCaoG I will be cherry-picking this PR in the 1.11 release branch and looks like it'll require an xla side change (remove `torch.complex32` from this list https://github.com/pytorch/xla/blob/3bb039c8e8cff30e23394e47bf67443b0c200100/test/pytorch_test_base.py#L467-L469)

JackCaoG commented:
@anjali411 Seems like we only need to remove this dtype from our repo, I can do that and backport to our 1.11 branch.

anjali411 committed:
Update on "Disable torch.complex32 dtype"

Related https://github.com/pytorch/pytorch/issues/69209#event-5722454832




cc ezyang gchanan anjali411 dylanbespalko mruberry @Lezcano nikitaved

Differential Revision: [D33954861](https://our.internmc.facebook.com/intern/diff/D33954861)

[ghstack-poisoned]

anjali411 committed:
Update on "Disable torch.complex32 dtype"

Related https://github.com/pytorch/pytorch/issues/69209#event-5722454832




cc ezyang gchanan anjali411 dylanbespalko mruberry @Lezcano nikitaved

Differential Revision: [D33954861](https://our.internmc.facebook.com/intern/diff/D33954861)

[ghstack-poisoned]

anjali411 committed:
Update on "Disable torch.complex32 dtype"

Related https://github.com/pytorch/pytorch/issues/69209#event-5722454832




cc ezyang gchanan anjali411 dylanbespalko mruberry @Lezcano nikitaved

Differential Revision: [D33954861](https://our.internmc.facebook.com/intern/diff/D33954861)

[ghstack-poisoned]

malfet commented:
@anjali411 correct me if I'm wrong, but https://github.com/pytorch/pytorch/commit/13b403b7144fb99235e8e9c17316274a5a04dc17 has been landed in 1.11, so we probably can remove the milestone (and close this PR) after xla change is landed

JackCaoG commented:
@malfet I just merge the pt/xla 1.11 pr. Pytroch/xla 1.11 branch is missing several important prs. I am thinking about recut the 1.11 from master which is a bit easier than me manually porting a bunch of pr at the same time. Will let you know if I do that.

anjali411 commented:
> @anjali411 correct me if I'm wrong, but [13b403b](https://github.com/pytorch/pytorch/commit/13b403b7144fb99235e8e9c17316274a5a04dc17) has been landed in 1.11, so we probably can remove the milestone (and close this PR) after xla change is landed

yup absolutely!
